docs_src/ir_semantics.md

```diff
@@ -673,6 +673,60 @@ body(i, loop_carry_data[, invariant_arg0, invariant_arg1, ...])
 Note that we currently inspect the `body` function to see what type of induction
 variable (`i` above) it accepts in order to pass an `i` value of that type.
 
+#### **`dynamic_counted_for`**
+
+Invokes a dynamic-trip count loop.
+
+**Syntax**
+
+```
+result = counted_for(init, trip_count, stride, body=<body>, invariant_args=<inv_args>)
+```
+
+**Types**
+
+Value        | Type
+------------ | ----
+`init`       | `T`
+`trip_count` | `bits[N], treated as unsigned`
```

Comment:
I realize we call it "trip_count" in counted_for, but it is not really trip count. It is a bound on the range of the  induction variable. So if you have trip_count=10, stride=2, the trip count is actually 5. Mind filing a github issue for this? thanks.

Reply:
The implementations of the unroll pass (https://github.com/google/xls/blob/main/xls/passes/unroll_pass.cc#L43), ir interpreter (https://github.com/google/xls/blob/main/xls/interpreter/ir_interpreter.cc#L219), and llvm jit (https://github.com/google/xls/blob/main/xls/jit/function_builder_visitor.cc#L349) all treat trip_count as the number of trips, where the maximum value of the induction variable is (trip_count - 1) * stride.

Reply:
You are right. I misread the interpreter implementation. Sorry about that!